### ISSUE TITLE
Talk feature gap analysis (Dec 2022)

### ISSUE THREAD

**Issue by nimishavijay** [nextcloud/spreed#8518]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

Feature gap analysis for Nextcloud Talk against other widely used communication tools as of December 2022. If there are any existing issues regarding the features mentioned here, feel free to add links :) cc @jancborchardt 

| Features | Microsoft Teams | Zoom | Google Meet | Slack | Discord | Talk |
|----------|-----------------|------|-------------|-------|---------|------|
| Silent messages | ❌ | \- | \- | ❌ | ❌ | ✔️ |
| Breakout rooms | ✔️ | ✔️ | ✔️ | ➕ (add-on or app) | 🤖 using a bot | 🚧 <https://github.com/nextcloud/spreed/issues/3487> |
| Recording | ✔️ | ✔️ | ✔️ | ❌ | ❌ | 🚧 <https://github.com/nextcloud/spreed/issues/8324> |
| View time zones of colleagues | ✔️ [Tutorial](https://www.youtube.com/watch?v=VY2KjAtG4NA) | ? | ? | ? | ? | ❌ <https://github.com/nextcloud/spreed/issues/4657> |
| Instant disposable meetings | ✔️ [Support post](https://support.microsoft.com/en-us/office/start-an-instant-meeting-in-teams-ff95e53f-8231-4739-87fa-00b9723f4ef5) | ✔️ | ✔️ | ✔️ | ✔️ | ❌ <https://github.com/nextcloud/spreed/issues/5972> |
| Search filters | ✔️ [Support post](https://support.microsoft.com/en-us/office/search-and-filter-messages-31682332-4dba-42e5-9e3b-bf7dd178d106) | \- | \- | ✔️ | ❌ | ❌ |
| Note to self | ✔️ [Article about it](https://office365itpros.com/2022/06/13/teams-chat-with-self/) | \- | \- | ✔️ "Jot something down" | ❌ | ❌ Possible but not created by default <https://github.com/nextcloud/spreed/issues/2196> |
| Quick slash commands | ✔️ [Support post](https://support.microsoft.com/en-us/office/use-commands-in-teams-88f61508-284d-417f-a53d-9e082164050b) | \- | \- | ✔️ | ✔️ | ❌ We have commands, just no UI hints/autocomplete |
| Bots | ✔️ [Developer docs about building bots](https://learn.microsoft.com/en-us/microsoftteams/platform/bots/what-are-bots) | \- | \- | ✔️ | ✔️ | ❌ <https://github.com/nextcloud/spreed/issues/1879> |
| Threads | ✔️ | \- | \- | ✔️ | ✔️ [Post](https://support.discord.com/hc/en-us/articles/4403205878423-Threads-FAQ) | ❌ <https://github.com/nextcloud/spreed/issues/1469> |
| Format messages | ✔️ | \- | \- | ✔️ | ✔️ | ❌ <https://github.com/nextcloud/spreed/issues/1027> |
| Live reactions in meetings | ✔️ [Post](https://support.microsoft.com/en-us/office/express-yourself-in-teams-meetings-with-live-reactions-a8323a40-3d07-4129-934b-305370a36e21) | ✔️ | Announced, not out yet | ❌ | [Announced not out yet](https://twitter.com/disexperiments/status/1547284675160317955?lang=en) | ❌ <https://github.com/nextcloud/spreed/issues/364> |
| Together mode | ✔️ [Post](https://techcommunity.microsoft.com/t5/microsoft-teams/enabling-together-mode-in-ms-teams/m-p/1698285) | ✔️ [Immersive view](https://blog.zoom.us/introducing-zoom-immersive-view/) | ❌ | ❌ | ❌ | ❌ |
| Change conversation during meeting | ✔️ [Post](https://support.microsoft.com/en-us/topic/01835df8-c655-435e-9ad8-d756caa2ba07) | \- | \- |  | ✔️ | ❌ Can be done with opening another tab? |

### Resources

[MS Teams support website](https://support.microsoft.com/en-us/teams) contains detailed tutorials on specific features

**Comment by sebastiaanveld** [comment]
Not sure if you want additional solutions in this already great comparison, but I would like to add Matrix (#809) to the list. I would say Teams is the reference solution here as being the biggest competitor for all mentioned solutions, not in the last place because it tends to lock customers into the email-Office-Teams frame. IMO "just" chasing Teams to get feature compliant will keep one always behind the curve while the (or at least one of its) missions of NC is the need to do things better to stay relevant. Teams is, in my experience, not so friendly to work with, but part of an ecosystem of MS solutions. I think only having the same feature set may not be enough to convince customers to move to other solutions. To get ahead of the curve Matrix may add the needed features to NC/Talk that Teams does not (yet) have. Matrix adds like the ability to interact with other instances just like a role email provides today. By adding that protocol that would make one less dependent on email which is one of the primary reason customer moved to the cloud (O365, Google). As long as customers are locked in by their email I do not think they see much reason to move. Of cource MS also could implement the Matrix protocol into their solution, I eventually I think they will, but I think the postpone that as long as possible to maintain their lockin on email users. What helps is the increasing interest of organizations to move to Matrix solutions for (private) chat. Being able to integrate NC/Talk into such solutions may help customer to decide to move.

**Comment by sunjam** [comment]
@sebastiaanveld I think you mean by Matrix is you'd like for Talk to support [Matrix Federation API](https://matrix.org/docs/spec/server_server/r0.1.4).
This would allows Talk server-to-server, but also to Matrix / Teamspeak / Gitter, etc.
- #5723 is also worth referencing.

**Comment by nickvergessen** [comment]
> View time zones of colleagues

* Has been implemented for 26+ with https://github.com/nextcloud/server/pull/35979

**Comment by nuokh** [comment]
In contrast to enterprise solutions like [Teams ](https://www.ip-insider.de/polycom-unterstuetzt-microsoft-teams-a-738038/) or [Zoom](https://www.poly.com/de/de/solutions/platform/zoom/video), **NC Talk is missing the ability to integrate with a hardware conference room solution** of any vendor, e.g. [Logitech](https://www.logitech.com/en-gb/products/video-conferencing/room-solutions.html) or [Polycom](https://www.poly.com/de/de/solutions/platform).

**Comment by nickvergessen** [comment]
> NC Talk is missing the ability to integrate with a hardware conference room solution of any vendor,

Any conference room system that works with a PC/Laptop/Tablet/mobile Phone is compatible.

Citrix was tested and is used successfully, and in our own office rooms we use devices like the one from Konftel: https://www.konftel.com/en/solutions/konftel-c50800-hybrid

**Comment by jancborchardt** [comment]
Please note this is not another place for feature requests – use the [issue templates for that](https://github.com/nextcloud/spreed/issues/new/choose). Here we want to have a quick overview of the market and competitors.

**Comment by MPStudyly** [comment]
Slack allows to view colleagues time zones by browsing their profile. That can be done by one click from PM view.
![image](https://user-images.githubusercontent.com/115081586/223700865-2de66ecc-a689-40d3-8447-258abc18cbdd.png)
Comment can be cleared/removed after adding this info.

**Comment by nickvergessen** [comment]
![grafik](https://user-images.githubusercontent.com/213943/223706169-c40ed8a2-c44d-4f60-b378-921c601036b4.png)

That's also there in Nextcloud 26 and later as per https://github.com/nextcloud/spreed/issues/8518#issuecomment-1371800538

**Comment by nickvergessen** [comment]
@nimishavijay want to check the state of "Drafts" across the other platforms?
Related Talk issue is https://github.com/nextcloud/spreed/issues/3055

**Comment by alexanderdd** [comment]
Some feedback. @nimishavijay what do you think?

- I think we should move Talk to the first column. (coloring it is not possible I guess)
- IMO, a comparison with BigBlueButton would make sense. In the education sector, my perception is that next to Zoom and MS Teams, BBB is the third biggest one, and people comparing options would like to know how Talk compares to them.
- There are so many features, maybe it is possible to group them into two or three categories to make it easier to grasp?

**Comment by jospoortvliet** [comment]
@alexanderdd this isn't a marketing thing, just something engineering uses to make decisions on features. I'm saying - column order doesn't matter ;-)

I'll update the PDF we made for marketing, for which we used (most of) this data - but it isn't the same. Your point about BBB is relevant, though. But somebody has to do the research before we can add it :/

**Comment by alexanderdd** [comment]
@nimishavijay I added BigBlueButton (last column before Talk). Can you check if my additions are clear, or if there are any questions?

**Comment by jancborchardt** [comment]
@alexanderdd btw, if you are interested in also looking into Mattermost and rocket.chat or have done so already, feel free to add them as well. :) cc @nimishavijay

**Comment by sunjam** [comment]
@jancborchardt it may be worth looking into what actually happened with the Nextcloud / Rocket.chat integration.  It immediately failed.
[See the github repo here](https://github.com/RocketChat/rocketchat_nextcloud) and [the official Nextcloud announcement here](https://nextcloud.com/blog/rocket-chat-nextcloud-integration/).  Rocket.chat has their support at https://open.rocket.chat but it seems no coordination ever took place as far as can be found publicly.

Not meaning to derail your thread, but since you've specifically asked about the rocket.chat platform it is worth looking into why integration efforts failed for another open source platform so clearly interested in Nextcloud support.  I asked them about it 1.5 years ago, but got no response.  Hope this helps glean more info, since I love both of these projects.

**Comment by alexanderdd** [comment]
Hello,

I recently looked at Zoom in more detail. I propose the following changes to the table (Nimisha told me to post it here, not to do the changes myself):

Rich interactive widgets 
-> NC: remove the link, or change link text to "latest features"

Instant disposable meetings 
-> change bold font to normal, change NC to ✔️

❓ Outlook integration to send invitations 
-> remove the ❓ 
-> Zoom change to ✔️ and add link https://support.zoom.com/hc/en/article?id=zm_kb&sysparm_article=KB0062420
-> does the "-" in the other cells mean "no" or "not checked yet"? Better use ✔️/❌/"not checked", fill all cells in this row.

Filter conversations
-> Zoom: change ❌ to ✔️: you can filter for mentions, bookmarks, files, reminders [actually it is not filtering the conversations but a separate view toggled by sidebar button, showing you e.g. all mentions across all conversations]

Change conversation during meeting
-> NC change to ✔️ , remove ? in comment [yes that is the way how you can be in multiple conversations, even calls, at the same time]
-> remove bold font

Message drafts
-> NC change ❌ to ✔️ (Draft preserved when using the same device. Drafts not synced across devices.)
-> Zoom change ❌ to ✔️ (I did not test this myself but this looks like proof https://community.zoom.com/t5/Zoom-Team-Chat/Draft-message-within-Channels/m-p/137863#M8856 )

Waiting music
-> NC should be ❌, remove "stylized ringing tone" [we don't have waiting music that participants can hear when waiting in the lobby, see linked issue]
-> Zoom should be ✔️  https://community.zoom.com/t5/Zoom-Meetings/Play-music-in-waiting-room/m-p/25237

Live translations
-> create an issue and link it here

Integration with calendar app
-> Zoom: add behind "calendar always available in sidebar" - also integration with M365, google, exchange (bidirectional sync)

Schedule meetings without leaving conversation
-> NC change ❌ to ✔️ 

Sidebar view in calls
-> not sure what this means? you can open left and right sidebar during a call. You can open a office document during a call in the main window, but also in the chat sidebar as interactive widget. 

Start a whiteboard in a call
-> NC change ❌ to ✔️ , remove bold font

# general comments

Zoom is not just a videoconferencimg software anymore, they try to be a full communication and collaboration solution. So in theory, they should be compared to Nextcloud Hub. I could help with that.

The biggest downside of Zoom is the weird UX around "permanent conversations". With NC, you just create conversations, and whenever you need, you start a call.

# next steps

Once these updates are integrated, maybe also update https://nextcloud.com/talk/#compare 

General idea: maybe it is good to link to the user docs for each NC feature? If not from here, then maybe from the https://nextcloud.com/talk/#compare

**Comment by alexanderdd** [comment]
To do:

- [ ] update the table above with features from NC32
- [ ] integrate my proposed changes from the message above ^

I can do this, but I need a 👍
